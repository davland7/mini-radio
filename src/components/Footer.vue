<template>
  <div class="footer">
    <div class="footer__version">
      {{ messages.version }}
    </div>
    <div class="footer__github">
      <a
        class="footer__link"
        :href="messages.homePageUrl"
        target="_blank"
      >
        {{ messages.github }}
      </a>
    </div>
  </div>
</template>

<script>
  export default {
    name: 'FooterComponent',
    props: {
      messages: {
        type: Object,
        required: true
      }
    }
  }
</script>

<style lang="scss">
  .footer {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 4px;
    margin: 4px 2px 2px;
    font-size: 10px;
    color: var(--dark-color);

    &__link {
      color: inherit;
      text-decoration: none;

      &:hover {
        text-decoration: underline;
      }
    }
  }
</style>
