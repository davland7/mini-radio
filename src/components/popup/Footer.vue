<template>
  <div class="footer">
    <div class="footer__version">
      {{messages.version}}
    </div>
    <div class="footer__github">
      <a @click="githubTrack" :href="messages.homePageUrl" target="_blank" class="footer__link">{{ messages.github }}</a>
    </div>
  </div>
</template>

<script>
  import GA from '../../googleAnalytics.js'

  export default {
    name: 'Footer',
    methods: {
      githubTrack () {
        GA.trackEvent(this.messages.github, 'clicked')
      }
    },
    data () {
      return {
        messages: {
          version: chrome.runtime.getManifest().version,
          homePageUrl: chrome.runtime.getManifest().homepage_url,
          github: chrome.i18n.getMessage('github')
        }
      }
    }
  }
</script>

<style lang="scss">
  .footer {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 4px;
    margin: 4px 2px 2px;
    background-color: #b34044;
    font-size: 10px;
    color: #fff;
  }

  .footer__link {
    color: #fff;
    text-decoration: none;

    &:hover {
      text-decoration: underline;
    }
  }
</style>
