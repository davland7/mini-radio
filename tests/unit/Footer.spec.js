import { shallowMount } from '@vue/test-utils'
import Footer from '@/components/Footer.vue'

const messages = {
  version: '2.0.0',
  homePageUrl: '#',
  github: 'GitHub'
}

describe('Footer.vue', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(Footer, {
      props: { messages }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
