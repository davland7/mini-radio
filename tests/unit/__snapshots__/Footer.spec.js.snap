// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.vue renders correctly 1`] = `
<div
  class="footer"
>
  <div
    class="footer__version"
  >
    2.0.0
  </div>
  <div
    class="footer__github"
  >
    <a
      class="footer__link"
      href="#"
      target="_blank"
    >
      GitHub
    </a>
  </div>
</div>
`;
